apps/antalmanac/src/components/RightPane/SectionTable/SectionTableBody.tsx

```diff
@@ -183,6 +200,100 @@ const InstructorsCell = withStyles(styles)((props: InstructorsCellProps) => {
 });
 
+async function getGpaData(deptCode: string, courseNumber: string, instructors: string[]) {
+    const namedInstructors = instructors.filter((instructor) => instructor !== 'STAFF');
+
+    // Get the GPA of the first instructor of this section where data exists
+    for (const instructor of namedInstructors) {
+        const grades = await Grades.queryGrades(
+            deptCode,
+            courseNumber,
+            instructor,
+            useTabStore.getState().activeTab != 1
+        );
+        if (grades?.averageGPA) {
+            return {
+                gpa: grades.averageGPA.toFixed(2).toString(),
+                instructor: instructor,
+            };
+        }
+    }
+
+    return undefined;
+}
+
+interface GPACellProps {
+    deptCode: string;
+    courseNumber: string;
+    instructors: string[];
+}
+
+function GPACell(props: GPACellProps) {
+    const { deptCode, courseNumber, instructors } = props;
+
+    const [gpa, setGpa] = useState('');
+
+    const [instructor, setInstructor] = useState('');
+
+    const [anchorEl, setAnchorEl] = useState<Element>();
+
+    const handleClick = useCallback((event: React.MouseEvent<HTMLElement>) => {
+        setAnchorEl((currentAnchorEl) => (currentAnchorEl ? undefined : event.currentTarget));
+    }, []);
+
+    const hideDistribution = useCallback(() => {
+        setAnchorEl(undefined);
+    }, []);
+
+    useEffect(() => {
+        getGpaData(deptCode, courseNumber, instructors)
+            .then((data) => {
+                if (data) {
+                    setGpa(data.gpa);
+                    setInstructor(data.instructor);
+                }
+            })
+            .catch(console.log);
+    }, [deptCode, courseNumber, instructors]);
+
+    return (
+        <NoPaddingTableCell>
+            <Button
+                style={{
+                    color: isDarkMode() ? 'dodgerblue' : 'blue',
+                    padding: 0,
+                    minWidth: 0,
+                    fontWeight: 400,
+                    fontSize: '1rem',
+                }}
+                onClick={handleClick}
+                variant="text"
+            >
+                {gpa}
+            </Button>
+            <Popover
+                open={Boolean(anchorEl)}
+                style={{
+                    color: isDarkMode() ? 'dodgerblue' : 'blue',
+                    cursor: 'pointer',
+                }}
+                onClose={hideDistribution}
+                anchorEl={anchorEl}
+                anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+                transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+                disableRestoreFocus
+            >
+                <GradesPopup
+                    deptCode={deptCode}
+                    courseNumber={courseNumber}
+                    instructor={instructor}
+                    isMobileScreen={useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT}`)}
```

Comment:
Same as before; maybe consider using MUI's built-in breakpoints.

```suggestion
                    isMobileScreen={useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT})`)}
```